src/record/api.rs

```diff
@@ -139,13 +142,28 @@ impl RowAccessor for Row {
 
 /// Constructs a `Row` from the list of `fields` and returns it.
 #[inline]
-pub fn make_row(fields: Vec<(String, Field)>) -> Row { Row { fields } }
+pub fn make_row(fields: Vec<(TypePtr, Field)>) -> Row { Row { fields } }
+
+impl PartialEq for Row {
+  fn eq(&self, other: &Row) -> bool {
```

Comment:
Equality is only used in tests, normally user would not run that in the production code.